src/CarbonAware.LocationSources/src/Configuration/LocationSourceFile.cs

```diff
@@ -0,0 +1,56 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.LocationSources.Configuration;
+
+public class LocationSourceFile
+{
+
+    private const string DefaultLocationDataFile = "azure-regions.json";
+    private const string BaseDirectory = "location-sources/json";
+    private const string DirectoryRegExPattern = @"^[-\\/a-zA-Z_\d ]*$";
+
+    private string assemblyDirectory;
```

Comment:
NIT: underscores for private member vars

```suggestion
    private string _assemblyDirectory;
```